website/versioned_docs/version-30.0/Configuration.md

```diff
@@ -1522,25 +1529,6 @@ const config: Config = {
 export default config;
 ```
 
-By combining `defaultResolver` and `packageFilter` we can implement a `package.json` "pre-processor" that allows us to change how the default resolver will resolve modules. For example, imagine we want to use the field `"module"` if it is present, otherwise fallback to `"main"`:
-
-```js
-module.exports = (path, options) => {
-  // Call the defaultResolver, so we leverage its cache, error handling, etc.
-  return options.defaultResolver(path, {
-    ...options,
-    // Use packageFilter to process parsed `package.json` before the resolution (see https://www.npmjs.com/package/resolve#resolveid-opts-cb)
-    packageFilter: pkg => {
-      return {
-        ...pkg,
-        // Alter the value of `main` before resolving the package
-        main: pkg.module || pkg.main,
-      };
-    },
-  });
-};
-```
-
```

Comment:
There is another way to do similar with new resolver:

```javascript
module.exports = (path, options) => {
  // Call the defaultResolver, so we leverage its cache, error handling, etc.
  return options.defaultResolver(path, {
    ...options,
    // HACK!!!
    // this is option from unrs-resolver from https://github.com/unrs/unrs-resolver?tab=readme-ov-file#main-field
    // unrs-resolver used from jest-resolve now https://github.com/jestjs/jest/blob/v30.0.0/packages/jest-resolve/src/defaultResolver.ts#L84-L98
    // We use the fact that jest-resolve just pass extra options to resolver
    mainFields: ["react-native", "main"],
  });
};
```

Reply:
that seems like a very reasonable way of doing it. should ve added to the docs 😀